src/Features/Core/Portable/SignatureHelp/SignatureContext.cs

```diff
@@ -0,0 +1,148 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.SignatureHelp
+{
+    internal sealed class SignatureContext
+    {
+        private readonly List<SignatureHelpItem> _items;
+        private TextSpan _applicableSpan;
+        private SignatureHelpState _state;
+
+        internal IReadOnlyList<SignatureHelpItem> Items
+        {
+            get { return _items; }
```

Comment:
FWIW, I'm not sure I need this at all. I inherited it from the `CompletionContext` code. I'll remove it if it's unneeded.